src/components/Search/SelectedRevisions.tsx

```diff
@@ -2,71 +2,72 @@ import { useEffect, useState } from 'react';
 
 import Box from '@mui/material/Box';
 import List from '@mui/material/List';
+import { useLocation } from 'react-router-dom';
 
-import { repoMap, searchView } from '../../common/constants';
+import { repoMap, searchView, compareView } from '../../common/constants';
 import { useAppSelector } from '../../hooks/app';
 import { SelectRevsStyles } from '../../styles';
 import {
   InputType,
-  View,
-  RevisionsList,
   Repository,
   ThemeMode,
+  RevisionsList,
 } from '../../types/state';
 import SelectedRevisionItem from './SelectedRevisionItem';
-
 interface SelectedRevisionsProps {
   mode: ThemeMode;
   searchType: InputType;
-  isWarning: boolean;
-  view: View;
-  editBtnVisible?: boolean;
+  isWarning?: boolean;
+  formIsDisplayed: boolean;
+  isEditable: boolean;
 }
 
 function SelectedRevisions({
   mode,
   searchType,
   isWarning,
-  view,
-  editBtnVisible,
+  formIsDisplayed,
+  isEditable,
 }: SelectedRevisionsProps) {
   const styles = SelectRevsStyles(mode);
+  const location = useLocation();
+  const view = location.pathname == '/' ? searchView : compareView;
   const [revisions, setRevisions] = useState<RevisionsList[]>([]);
   const [repositories, setRepositories] = useState<Repository['name'][]>([]);
   const checkedRevisionsList = useAppSelector(
     (state) => state.search[searchType].checkedRevisions,
   );
 
-  const selectedRevisions = useAppSelector(
-    (state) => state.selectedRevisions.revisions,
-  );
-
-  const displayedSelectedRevisions = useAppSelector(
-    (state) => state.selectedRevisions[searchType],
-  );
-
   const checkedRepositories = checkedRevisionsList.map((item) => {
     const selectedRep = repoMap[item.repository_id];
     return selectedRep;
   });
-
-  const selectedRevRepo = selectedRevisions.map((item) => {
+  const selectedRevisions = useAppSelector(
+    (state) => state.selectedRevisions.revisions,
+  );
+  const compareViewRepositories = selectedRevisions.map((item) => {
     const selectedRep = repoMap[item.repository_id];
     return selectedRep;
   });
+  const compareViewSelectedRevisions = useAppSelector(
+    (state) => state.selectedRevisions[searchType],
+  );
 
   useEffect(() => {
-    if (view === searchView || !editBtnVisible) {
+    if (view === searchView) {
```

Comment:
This one too, but I have an idea for this one: the revisions and repositories should be passed as props.